packages/ui/src/proposals/components/ProposalStages.tsx

```diff
@@ -3,44 +3,68 @@ import React, { useMemo } from 'react'
 import { ControlProps } from '@/common/components/forms'
 import { CheckboxIcon, CrossIcon } from '@/common/components/icons'
 import { TabContainer, TabsContainer } from '@/common/components/Tabs'
+import { Tooltip } from '@/common/components/Tooltip'
 import { last, repeat } from '@/common/utils'
 import { ProposalStatus, ProposalStatusUpdates } from '@/proposals/types'
 
 export interface ProposalStagesProps extends ControlProps<number> {
   status: ProposalStatus
   updates: ProposalStatusUpdates[]
-  constitutionality?: number | string
+  constitutionality?: number
 }
 
-export const ProposalStages = ({ status, updates, constitutionality = '-', value, onChange }: ProposalStagesProps) => {
+const iconMap = { approved: <CheckboxIcon />, rejected: <CrossIcon />, deciding: undefined, dormant: undefined }
+
+export const ProposalStages = ({ status, updates, constitutionality, value, onChange }: ProposalStagesProps) => {
   const rounds = useMemo(() => {
     const decidingCount = updates.filter(({ status }) => status === 'deciding').length
+
+    const totalRoundCount = typeof constitutionality === 'string' ? undefined : constitutionality
+
     const lastUpdate = last(updates).status
     const onGoing = lastUpdate === status
     const approved = lastUpdate === 'gracing'
     const rejected = !onGoing && !approved
     const isDeciding = onGoing && status === 'deciding'
     const isDormant = onGoing && status === 'dormant'
+
     return [
-      ...repeat((round) => ({ icon: <CheckboxIcon />, onClick: () => onChange(round) }), decidingCount - 1),
-      {
-        icon: !isDeciding && (rejected ? <CrossIcon /> : <CheckboxIcon />),
-        onClick: () => onChange(decidingCount - 1),
-      },
-      ...(isDormant ? [{ icon: false, onClick: undefined }] : []),
+      ...(isDormant
+        ? repeat(
+          () => (status),
+          constitutionality ?? 1
+        )
+        : []),
     ]
-  }, [updates.length, status])
+  }, [updates.length, status, constitutionality])
+
 
   return (
     <TabsContainer>
-      {rounds.map(({ icon, onClick }, round) => (
-        <TabContainer key={round} active={round === value} disabled={!onClick} onClick={onClick}>
-          {icon}
-          <span>
-            Council approvals {round + 1}/{constitutionality}
-          </span>
-        </TabContainer>
-      ))}
+      {rounds.map((status, round) => {
+        const isDisabled = status === 'dormant'
```

Comment:
Referring to the [original comment](https://github.com/Joystream/pioneer/pull/4119#discussion_r1091047618): `dormant` is the phase after approval until the next council. However if a tab `isDisabled` depends on content of `onClick`:
> Here the "disabled" tabs are in fact those with typeof onClick === 'undefined' (not round !== value), which is not very readable code.